ubloxgps/gpsparser.go

```diff
@@ -0,0 +1,137 @@
+package ubloxgps
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type GPSParser struct {
+	gpsDevice GPSDevice
+}
+
+// fix is a GPS location fix
+type Fix struct {
+	Valid     bool
+	Time      string
+	Latitude  string
+	Longitude string
+	Altitude  int32
+	Satelites int16
+}
+
+func Parser(gpsDevice GPSDevice) GPSParser {
+	return GPSParser{
+		gpsDevice: gpsDevice,
+	}
+}
+
+func (parser *GPSParser) NextFix(fix *Fix) {
+	// println("ReadNextFix")
+	var ggaSentence = nextGGA(parser.gpsDevice)
+	var ggaFields = strings.Split(ggaSentence, ",")
+	fix.Valid = true
+	fix.Altitude = findAltitude(ggaFields)
+	fix.Satelites = findSatelites(ggaFields)
+	fix.Longitude = findLongitude(ggaFields)
+	fix.Latitude = findLatitude(ggaFields)
+	fix.Time = findTime(ggaFields)
+}
+
+func nextGGA(gpsDevice GPSDevice) (sentence string) {
+	// println("ReadNextGGA")
+	// $--GGA,,,,,,,,,,,,,,*hh
```

Comment:
This line

```
// $--GGA,,,,,,,,,,,,,,*hh
```

probably should be part of the comment for the `nextGGA()` function.